docker-build.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# This program will read a list of Scala and SBT versions from build config
+# files. For each pair of versions, it will determine if a version already exists
+# on docker hub for that version. If it does it skips that version.
+
+# A FORCE env var can be set to 1 (one) in order to force the building of all version
+# pairs of Scala and SBT regardless of whether or not they already exist on dockerhub.
+# This is useful to ensure the docker image has the newest JDK version since currently
+# only the major version (8) is specified for JDK and the script will get whatever the
+# latest version 8 of JDK exists at the time of build.
+
+FORCE=${FORCE:-0}
+
+# Docker Hub container name
+REPO="broadinstitute/scala"
+
+# Hard-code to JDK 8
+JDK_VERSION=8
+
+# Files containing version lists
+SCALA_CONFIG="scala-list.cfg"
+SBT_CONFIG="sbt-list.cfg"
+
+# Generic error outputting function
+errorout() {
+   if [ $1 -ne 0 ];
+        then
+        echo "${2}"
+        exit $1
+    fi
+}
+
+for config in "$SCALA_CONFIG" "$SBT_CONFIG"; do
+  if [ -f "$config" ]; then
+    if [ ! egrep -q "^[a-zA-Z0-9]+" "$config" ]; then
+      errorout 1 "No valid entries in config file: ${config}"
+    fi
+  else
+    errorout 1 "Missing version list: ${config}"
+  fi
+done
+
+egrep "^[a-zA-Z0-9]+" "$SCALA_CONFIG" | while read scala_version rest; do
+  egrep "^[a-zA-Z0-9]+" "$SBT_CONFIG" | while read sbt_version rest; do
+    docker_version="scala-${scala_version}-sbt-${sbt_version}"
+    build_version="$FORCE"
+
+    # Ensure Dockerfile doesn't exist
+    rm -f Dockerfile
+
+    echo "Checking if Scala version (${scala_version}) on SBT version (${sbt_version}) needs to be built..."
+    if [ "$FORCE" -ne "1" ]; then
+      # See if version exists on docker hub
+      docker pull ${REPO}:${docker_version}
+      retcode=$?
+
+      if [ "$retcode" -eq "0" ]; then
+        echo "Skipping version"
+        build_version=0
+      else
+        build_version=1
+      fi
```

Comment:
Instead of repeating `else build_version=1` twice, I changed `build_version` to default to 1.